### PR TITLE
Gasping for words

### PR DESCRIPTION
**Pull request by tedwood3**
When the oxygen alert is active, there is now a 50% chance that a player will stop half way through their sentence and gasp for words.
This somewhat addresses the [suggestion](https://forums.aurorastation.org/viewtopic.php?f=18&t=10554) posted on the forum by [Fortport](https://forums.aurorastation.org/memberlist.php?mode=viewprofile&u=279)

image

### CODE REVIEW

**Comment by BurgerLUA** [comment]
It's like you're trying to give me depression which each passing miner's death.
This is good.

**Comment by tedwood3** [comment]
Sorry. Still getting the hang of git. I've reset to development, applied my changes and pushed that to my GASP branch.

**Comment by Lohikar** [comment]
Probably better to base this off losebreath instead of the UI element. That would in theory make this apply to people winded from sprinting too much too.

File: code/modules/mob/living/carbon/human/say.dm [hunk 1]
```diff
@@ -297,4 +297,24 @@
 				if(length(cword))
 					rearranged += cword
 			message ="[prefix][jointext(rearranged," ")]"
+		if(oxygen_alert) //Gasping is a mutation, right?
```
Comment (skull132) [inline code/modules/mob/living/carbon/human/say.dm]: Lohikar is correcto.

Check the variable `losebreath` instead. It's increased every time a person is unable to breathe. For reference, some chemicals use the condition of `if (losebreath >= 10)`, though you'll need to see where the nominal point is.

File: code/modules/mob/living/carbon/human/say.dm [hunk 2]
```diff
@@ -297,4 +297,24 @@
 				if(length(cword))
 					rearranged += cword
 			message ="[prefix][jointext(rearranged," ")]"
+		if(oxygen_alert) //Gasping is a mutation, right?
+			var/prefix=copytext(message,1,2)
+			if(prefix == ";")
+				message = copytext(message,2)
+			else if(prefix in list(":","#"))
```
Comment (skull132) [inline code/modules/mob/living/carbon/human/say.dm]: doesn't the user also have the ability to set custom prefixes for certain things? Or was it only for languages, ergo it doesn't matter.

Comment (tedwood3) [reply code/modules/mob/living/carbon/human/say.dm]: That's the same code used for tourettes syndrome. It must be the latter

**Comment by tedwood3** [comment]
Accidentally wiped my work. Brought it back somehow. It is what it is.
I changed the oxygen_alert to losebreath, as requested.

**Comment by skull132** [comment]
Technically, might want to do `if (losebreath > 5)` or something. As it's really easy to just lose breath once or twice and then be fine. But your call, as this isn't auto-triggered and is instead triggered upon speech.

**Comment by tedwood3** [comment]
I took your advice and changed losebreath to losebreath >= 5. Good idea, @skull132 
I was looking into adding paincrit, but I can't find the variable for it.

**Comment by skull132** [comment]
@tedwood3 if you want pain (HALLOSS) to also affect this, compare `halloss > someamount`. Something like > 30 should do the trick.

**Comment by chaoko99** [comment]
breathe, not breath.

**Comment by skull132** [comment]
You don't lose your breathe, @chaoko99 . You lose your breath.

**Pull request merged**